src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/resource/ActivitiesResourceIntTest.kt

```diff
@@ -49,31 +55,37 @@ class ActivitiesResourceIntTest : IntegrationTestBase() {
   @Autowired
   lateinit var repository: Repository
 
-  @DisplayName("Create")
-  @Nested
-  inner class Create {
-    @BeforeEach
-    internal fun createProgramService() {
-      repository.save(
-        ProgramService(
-          programId = 20,
-          programCode = programCode,
-          description = "test description",
-          active = true,
-        )
+  lateinit var offenderAtMoorlands: Offender
+
+  @BeforeEach
+  internal fun setup() {
+    repository.save(
+      ProgramService(
+        programId = 20,
+        programCode = programCode,
+        description = "test description",
+        active = true,
       )
-    }
+    )
+    offenderAtMoorlands =
+      repository.save(OffenderBuilder(nomsId = "A1234TT").withBooking(OffenderBookingBuilder(agencyLocationId = "MDI")))
+  }
 
-    @AfterEach
-    internal fun cleanUp() {
-      repository.deleteActivities()
-      repository.deleteProgramServices()
-    }
+  @AfterEach
+  internal fun cleanUp() {
+    repository.deleteOffenderProgramProfiles()
+    repository.delete(offenderAtMoorlands)
+    repository.deleteActivities()
+    repository.deleteProgramServices()
+  }
```

Comment:
again, feels like deleting an activity would delete the child relationships, are missing a relationship ?

Reply:
added 2 bidirectional relationships